Physical_Engine/Query/Geometry.cs

```diff
@@ -59,6 +62,41 @@ public static oM.Geometry.ISurface Geometry(this oM.Physical.Elements.ISurface s
 
         /***************************************************/
 
+        [Description("Gets the geometry of a PadFoundation as a surface. Method required for automatic display in UI packages.")]
+        [Input("padFoundation", "PadFoundation to get the surface geometry from.")]
+        [Output("surface", "The surface defining the PadFoundation.")]
+        public static IGeometry Geometry(this PadFoundation padFoundation)
+        {
+            return padFoundation.IsNull() ? null : padFoundation.Location;
```

Comment:
Same as below, it is in general better to simply check if the value is `null` in `Geometry` queries rather than calling `IsNull` to avoid irrelevant errors.
```suggestion
            return padFoundation?.Location;
```